src/prefect/runner/storage.py

```diff
@@ -276,6 +279,137 @@ def to_pull_step(self) -> Dict:
         return pull_step
 
 
+class RemoteStorage:
+    def __init__(
+        self,
+        url: str,
+        pull_interval: Optional[int] = 60,
+        **settings: Any,
+    ):
+        self._url = url
+        self._settings = settings
+        self._logger = get_logger("runner.storage.remote-storage")
+        self._storage_base_path = Path.cwd()
+        self._pull_interval = pull_interval
+
+    @staticmethod
+    def get_required_package_for_scheme(scheme: str) -> Optional[str]:
+        # attempt to discover the package name for the given scheme
+        # from fsspec's registry
+        known_implementation = fsspec.registry.get(scheme)
+        if known_implementation:
+            return known_implementation.__module__.split(".")[0]
+        # if we don't know the implementation, try to guess it for some
+        # common schemes
+        elif scheme == "s3":
+            return "s3fs"
+        elif scheme == "gs" or scheme == "gcs":
+            return "gcsfs"
+        elif scheme == "abfs" or scheme == "az":
+            return "adlfs"
+        else:
+            None
+
+    @property
+    def filesystem(self) -> fsspec.AbstractFileSystem:
+        scheme, _, _, _, _ = urlsplit(self._url)
+
+        def replace_blocks_with_values(obj: Any) -> Any:
+            if isinstance(obj, Block):
+                if hasattr(obj, "get"):
+                    return obj.get()
+                if hasattr(obj, "value"):
+                    return obj.value
+                else:
+                    return obj.dict()
+            return obj
+
+        settings_with_block_values = visit_collection(
+            self._settings, replace_blocks_with_values, return_data=True
+        )
+
+        return fsspec.filesystem(scheme, **settings_with_block_values)
```

Comment:
Could there be an issue with settings here where we also want error handling here?

Reply:
I don't think it will raise due to a settings issue until you attempt to pull the code. The `fsspec` errors are generally pretty helpful because they'll let you know which package you need to install if it's missing.

Reply:
ah, nice 👍